### PR TITLE
Moving rqt to pure python

### PR DESCRIPTION
**Pull request by brawner**
To enable bundling of python scripts (rqt, rqt_gui) into executables on Windows and to significantly speed up build time, this PR converts rqt, rqt_gui and rqt_gui_py to ament_python packages.

This addresses issue https://github.com/ros-visualization/rqt/issues/166

### CODE REVIEW

File: rqt/setup.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from setuptools import setup
+
+package_name = 'rqt'
+setup(
+    name=package_name,
+    version='0.5.0',
+    install_requires=['rqt_gui', 'rqt_gui_cpp', 'rqt_gui_py', 'rqt_py_common'],
+    data_files=[
+        ('share/ament_index/resource_index/packages',
+            ['resource/' + package_name]),
```
Comment (mlautman) [inline rqt/setup.py]: is this actually needed? launch is a pure python package and it doesn't do this empty file thing

File: rqt_gui/setup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+package_name = 'rqt_gui'
+setup(
+    name=package_name,
+    version='0.5.0',
+    package_dir={'': 'src'},
+    packages=['rqt_gui'],
+    data_files=[
+        ('share/ament_index/resource_index/packages',
+            ['resource/' + package_name]),
```
Comment (mlautman) [inline rqt_gui/setup.py]: same as above

File: rqt_gui/setup.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+from setuptools import setup
+
+package_name = 'rqt_gui'
+setup(
+    name=package_name,
+    version='0.5.0',
+    package_dir={'': 'src'},
+    packages=['rqt_gui'],
+    data_files=[
+        ('share/ament_index/resource_index/packages',
```
Comment (mlautman) [inline rqt_gui/setup.py]: Why does this work on Windows?? Shouldn't it be `share\ament_index\...`?

File: rqt/setup.py (hunk 1 above)
Comment (brawner) [reply rqt/setup.py]: You're right, this is no longer needed it seems.

File: rqt_gui/setup.py (hunk 2 above)
Comment (brawner) [reply rqt_gui/setup.py]: Doesn't look like this is needed.

File: rqt_gui/setup.py (hunk 3 above)
Comment (brawner) [reply rqt_gui/setup.py]: setuptools takes care of the cross-platform bit of this code.

**Comment by dirk-thomas** [comment]
The Python packages need to register themselves at the resource index as well as install their manifest file: see https://github.com/ros2/demos/blob/f2a1d9f5fe694a11fb42f3c036dacd130141ac0e/demo_nodes_py/setup.py#L10-L14 for an example.

Also since the patch removed all CMake based linters the Python packages need linter tests instead: see https://github.com/ros2/demos/tree/f2a1d9f5fe694a11fb42f3c036dacd130141ac0e/demo_nodes_py/test

The information in the `setup.py` files should match the existing information in the `package.xml` files: e.g. description, license, authors.

I haven't checked the entry points myself but it looks weird that they reference a `main` function in the `rqt_gui.main` module which doesn't exist - only a `Main` and a check for `__main__`?

**Comment by dirk-thomas** [comment]
Was this removed from the target branch after being merged?

**Comment by brawner** [comment]
@dirk-thomas It's been reopened as #170. For an unknown reason Mike can't build the package as it is.

**Pull request merged**